src/OrbitGl/Viewport.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "Viewport.h"
+
+#include <algorithm>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_gl {
+
+Viewport::Viewport(int width, int height) : width_(width), height_(height) {
+  world_width_ = width;
+  world_height_ = height;
+  world_extents_ = Vec2(width, height);
+}
+
+void Viewport::Resize(int width, int height) {
+  CHECK(width > 0);
+  CHECK(height > 0);
+
+  if (width == width_ && height == height_) return;
+
+  width_ = width;
+  height_ = height;
+  FlagAsDirty();
+}
+
+int Viewport::GetWidth() const { return width_; }
```

Comment:
Nit: Those one-liners can be moved to the header.

Reply:
I prefer to have them in the cpp - I assume this class will be included quite a lot at some point, and we already suffer from compilation times whenever we change anything in GlCanvas.h, so I'd like to keep the potential for header changes minimal here.